modules/admin/src/test/kotlin/jp/co/soramitsu/iroha2/engine/IrohaRunnerExtension.kt

```diff
@@ -0,0 +1,66 @@
+package jp.co.soramitsu.iroha2.engine
+
+import jp.co.soramitsu.iroha2.AdminIroha2Client
+import jp.co.soramitsu.iroha2.testcontainers.IrohaContainer
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.extension.InvocationInterceptor
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext
+import java.lang.reflect.Method
+import kotlin.reflect.KMutableProperty1
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.declaredMemberProperties
+
+class IrohaRunnerExtension : InvocationInterceptor {
```

Comment:
I think you are right - better to move this to `testcontainer`. In that case `admin` can be imported there. Below you can find method `setPropertyValue` to set value if target class has property with that type. We can make some changes to support both `Iroha2Client` and `AdminIroha2Client`